arcaflow_plugin_coremark_pro/coremark_pro_plugin.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+import subprocess
+import sys
+import typing
+from arcaflow_plugin_sdk import plugin
+from coremark_pro_schema import (
+    CertifyAllParams,
+    certifyAllResultSchema,
+    SuccessOutput,
+    ErrorOutput,
+)
+
+
+def run_oneshot_cmd(command_list, workdir) -> str:
+    try:
+        cmd_out = subprocess.check_output(
+            command_list, stderr=subprocess.PIPE, text=True, cwd=workdir
+        )
+    except subprocess.CalledProcessError as error:
+        return "error", ErrorOutput(
+            f"{error.cmd[0]} failed with return code "
+            f"{error.returncode}:\n{error.output}"
+        )
+    return "completed", cmd_out
+
+
+@plugin.step(
+    id="certify-all",
+    name="certify-all",
+    description=(
+        "Runs all of the nine tests, collects their output scores, and processes them "
+        "to generate the final CoreMark-PRO score"
+    ),
+    outputs={"success": SuccessOutput, "error": ErrorOutput},
+)
+def certify_all(
+    params: CertifyAllParams,
+) -> typing.Tuple[str, typing.Union[SuccessOutput, ErrorOutput]]:
+
+    ca_cmd = [
+        "make",
+        "-s",
+        "certify-all",
+        f"XCMD='-c{params.contexts} -w{params.workers}'",
+    ]
```

Comment:
A make command to run the tool? Odd ..

Reply:
Indeed. But re-implementing the complex makefile "script" and how it passes variables and post-processes results would be far too much investment. For the benchmark, `make certify-all` is the given and accepted way to run the suite, so I just stuck with it.